dev_tools/requirements/re-pip-compile-in-docker.sh

```diff
@@ -0,0 +1,31 @@
+#!/bin/bash
+
+#  Copyright 2023 Google LLC
+#
+#  Licensed under the Apache License, Version 2.0 (the "License");
+#  you may not use this file except in compliance with the License.
+#  You may obtain a copy of the License at
+#
+#      https://www.apache.org/licenses/LICENSE-2.0
+#
+#  Unless required by applicable law or agreed to in writing, software
+#  distributed under the License is distributed on an "AS IS" BASIS,
+#  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+#  See the License for the specific language governing permissions and
+#  limitations under the License.
+
+set -ex
+
+# We use docker to run the actual `pip-compile` command because its
+# behavior depends on the platform from which it is run. Please see
+# ./Dockerfile for the list of `pip-compile` commands that are run.
+docker build -t qualtran-pip-compile .
+
+# Create a container from the image so we can copy out the outputs
+id=$(docker create qualtran-pip-compile)
```

Comment:
FWIW `id` is a bash command already. What about image_id?

Reply:
fun fact this is a container id and _not_ an image id (I learned that those are different things and why this line is required at all :) )

Reply:
lol great point!! an even better reason to name it =D

Reply:
done